qt-models/divetripmodel.cpp

```diff
@@ -222,6 +222,7 @@ QVariant DiveTripModelBase::diveData(const struct dive *d, int column, int role)
 	case MobileListModel::SumWeightRole: return get_weight_string(weight_t { total_weight(d) }, true);
 	case MobileListModel::DiveMasterRole: return d->divemaster ? d->divemaster : QString();
 	case MobileListModel::BuddyRole: return d->buddy ? d->buddy : QString();
+	case MobileListModel::TagsRole: return d->tag_list ? get_taglist_string(d->tag_list) : QString();
```

Comment:
`get_taglist_string()` should support empty lists. If it doesn't, let's fix `get_taglist_string()`

Reply:
it does. But it's painfully convoluted. It creates a new `struct membuffer`, allocates one byte for the trailing `'\0'`, hands that 'C-string' back to an intermediary function, which then converts it to a `QString` and returns it to us.
But yeah, whatever. It works. Or I could move the `NULL` test one level down into `get_taglist_string()`... I'll take a look.

Reply:
after reading through this some more and looking at other callers I decided to file this under 'pointless optimizations' and moved on